### PR TITLE
Remove conditional now that 'Back to Sites' is translated

### LINKED ISSUE #67183: Tidy `hasTranslation` checks in "Back to Sites" label
Once the strings added in #67170 have been translated to all mag16 languages we should remove the `hasTranslation` checks and use `translate( 'Back to Sites' )` directly in the JSX like we usually do.

The translation bot should notify us when translations for the PR are complete.

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes #67183

## Proposed Changes

Removes the conditional now that 'Back to Sites' [is translated](https://translate.wordpress.com/projects/wpcom/-all-translated/249622/).

## Testing Instructions

1. Navigate to `/sites`.
2. Click on 'Add new site' in the top right.
3. Observe 'Back to Sites' in the top left.

image

### CODE REVIEW

**Review by vykes-mac — COMMENTED** [review]
good to go after merge conflicts resolved

**Review by vykes-mac — APPROVED** [review]

**Review by wojtekn — APPROVED** [review]
Looks good and works as expected.

**Review by zaguiini — APPROVED** [review]
Great clean up 🧹

**Pull request merged**
